contracts/external/cw-vesting/src/contract.rs

```diff
@@ -0,0 +1,556 @@
+        msgs.push(transfer_msg);
+    }
+
+    // Get owner's payout message in event of canceled and unbonding
+    if vesting_payment.status == VestingPaymentStatus::CanceledAndUnbonding {
+        let unvested_amount = vesting_payment.amount.checked_sub(vested_amount)?;
+        let owner = cw_ownable::get_ownership(deps.storage)?;
+        if let Some(owner) = owner.owner {
+            if unvested_amount != Uint128::zero() {
+                let withdraw_unbonded_msg = vesting_payment
+                    .denom
+                    .get_transfer_to_message(&owner, unvested_amount)?;
+                msgs.push(withdraw_unbonded_msg);
+            }
+        }
+    }
+
+    Ok(Response::new()
+        .add_attribute("method", "distribute")
+        .add_attribute("vested_amount", vested_amount)
+        .add_messages(msgs))
+}
+
+pub fn execute_update_owner(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+    action: cw_ownable::Action,
+) -> Result<Response, ContractError> {
+    let ownership = cw_ownable::update_ownership(deps, &env.block, &info.sender, action)?;
+    Ok(Response::default().add_attributes(ownership.into_attributes()))
+}
+
+pub fn execute_delegate(
+    _env: Env,
+    deps: DepsMut,
+    info: MessageInfo,
+    validator: String,
+    amount: Uint128,
+) -> Result<Response, ContractError> {
+    let vesting_payment = VESTING_PAYMENT.load(deps.storage)?;
+
+    // Check sender is the recipient of the vesting payment
+    if info.sender != vesting_payment.recipient {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Non-payable as this method delegates funds in the contract
+    nonpayable(&info)?;
+
+    // Check status is active
+    if vesting_payment.status != VestingPaymentStatus::Active {
+        return Err(ContractError::NotActive);
+    }
+
+    // Check vesting payment denom matches local denom
+    let denom = match vesting_payment.denom {
+        CheckedDenom::Cw20(_) => {
+            return Err(ContractError::NotStakeable);
+        }
+        CheckedDenom::Native(denom) => {
+            // Get local denom, ensure it matches the vesting denom
+            let local_denom = deps.querier.query_bonded_denom()?;
+            if local_denom != denom {
+                return Err(ContractError::NotStakeable);
+            } else {
+                denom
+            }
+        }
+    };
+
+    // Update vesting payment with amount of staked rewards
+    VESTING_PAYMENT.update(deps.storage, |mut vp| -> Result<_, ContractError> {
+        // Check stake amount is not greater than vesting payment amount
+        if amount > vp.amount {
+            return Err(ContractError::NotEnoughFunds {});
+        }
+
+        // Check that amount has not already been staked
+        vp.amount.checked_sub(vp.staked_amount + amount)?;
+
+        // Update amounts
+        vp.staked_amount += vp.staked_amount.checked_add(amount)?;
```

Comment:
```suggestion
        vp.staked_amount = vp.staked_amount.checked_add(amount)?;
```
unless i'm missing something pretty fundamental here? can we write a test to catch this?

Reply:
No, this is an embarrassing mistake on my part.

Reply:
happens to everyone haha